src/osvr/Client/RenderManagerConfig.cpp

```diff
@@ -0,0 +1,190 @@
+/** @file
+    @brief OSVR rendermanager configuration
+
+    @date 2015
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com>
+
+*/
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/Client/RenderManagerConfig.h>
+#include <osvr/Common/ClientContext.h>
+#include <osvr/Util/Verbosity.h>
+
+// Library/third-party includes
+#include <json/value.h>
+#include <json/reader.h>
+
+namespace osvr {
+namespace client {
+
+    RenderManagerConfigPtr RenderManagerConfigFactory::create(OSVR_ClientContext ctx) {
+        try {
+            auto const configString = ctx->getStringParameter("/display");
+            RenderManagerConfigPtr cfg(new RenderManagerConfig(configString));
+            return cfg;
+        }
+        catch (std::exception const &e) {
+            OSVR_DEV_VERBOSE(
+                "Couldn't create a render manager config internally! Exception: "
+                << e.what());
+            return RenderManagerConfigPtr{};
+        }
+        catch (...) {
+            OSVR_DEV_VERBOSE("Couldn't create a display config internally! "
+                "Unknown exception!");
+            return RenderManagerConfigPtr{};
+        }
+    }
+
+    RenderManagerConfig::RenderManagerConfig(const std::string& render_manager_config)
+    {
+        parse(render_manager_config);
+    }
+
+    void RenderManagerConfig::parse(const std::string& render_manager_config)
+    {
+      Json::Reader reader;
+      Json::Value root;
+      reader.parse(render_manager_config, root, false);
+
+      m_directMode = root["render_manager_parameters"]["direct_mode"].asBool();
+      m_displayIndex = root["render_manager_parameters"]["direct_display_index"].asUInt();
+      m_directHighPriority = root["render_manager_parameters"]["direct_high_priority"].asBool();
```

Comment:
prefer something like this:

```c++
auto & params = root["render_manager_parameters"];
m_directMode = params["direct_mode"].asBool();
m_displayIndex = params["direct_display_index"].asUInt();
m_directHighPriority = params["direct_high_priority"].asBool();
```

Now, there's no error checking anywhere here, and the code all assumes that every value is present. If you want to access something with a default value, `params.get("direct_mode", false).asBool();` is the general idea.

(And the schema is less than ideal: names aren't camel case (standard for javascript and thus json), minimal use of nesting, but that's less of a code problem than a config file design problem)